cidemipyme-gatsby/src/components/sections/ContactSection.jsx

```diff
@@ -0,0 +1,227 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      emailText: '',
+      subjectText: '',
+      messageText: '',
+      showErrors: false,
+      showSentMessage: false,
+    };
+  }
+
+  onChangeInput(inputType, event) {
+    if (inputType === 'email') {
+      this.setState({
+        emailText: event.target.value,
+        showSentMessage: false,
+      });
+    } else if (inputType === 'subject') {
+      this.setState({
+        subjectText: event.target.value,
+        showSentMessage: false,
+      });
+    } else if (inputType === 'message') {
+      this.setState({
+        messageText: event.target.value,
+        showSentMessage: false,
+      });
+    }
+  }
+
+  onSubmitMessage(event) {
+    if (this.state.emailText.length === 0
+      || this.state.subjectText.length === 0
+      || this.state.messageText.length === 0
+    ) {
+      this.setState({
+        showErrors: true
+      });
+      event.preventDefault();
+    } else {
+      this.setState({
+        showSentMessage: true,
+        showErrors: false
+      });
+    }
+  }
+
+  renderMailToForm() {
+    let emailFeedback =
+      this.state.showErrors && this.state.emailText.length === 0
+        ? <FormFeedback>Escribe un correo personal válido.</FormFeedback>
+        : null;
+    let subjectFeedback =
+      this.state.showErrors && this.state.subjectText.length === 0
+        ? <FormFeedback>Escribe un asunto para mostrar en el correo.</FormFeedback>
+        : null;
+    let messageFeedback =
+      this.state.showErrors && this.state.messageText.length === 0
+        ? <FormFeedback>Redacta el cuerpo del correo aquí.</FormFeedback>
+        : null;
+    let sentMessage = this.state.showSentMessage
+        ? <FormText>
+           {'Recibirás en tu correo una copia del mensaje. Respondemos en 24 horas promedio.'}
+          </FormText>
+        : null;
+
+    return (
+      <Form className="contact-form" method="post" action={this.props.script} target="_blank">
+        <FormGroup>
+          <Label for="senderEmail">Correo</Label>
+          <Input
+            className="form-control p-3 rounded-0"
+            type="email"
+            name="email"
+            id="senderEmail"
+            valid={
+              !this.state.showErrors || this.state.emailText.length > 0
+              ? null : false
+            }
+            onChange={this.onChangeInput.bind(this, 'email')}
+            value={this.state.emailText}
+            placeholder="Correo electrónico" />
+          {emailFeedback}
+        </FormGroup>
+        <FormGroup>
+          <Label for="senderSubject">Asunto</Label>
+          <Input
+            className="form-control p-3 rounded-0"
+            type="text"
+            name="subject"
+            id="senderSubject"
+            valid={
+              !this.state.showErrors || this.state.subjectText.length > 0
+              ? null : false
+            }
+            onChange={this.onChangeInput.bind(this, 'subject')}
```

Comment:
you can create a function on the constructor:
```javascript
this.onChangeSubject = this.onChangeInput.bind(this, 'subject');
```

I read that it is not good to bind functions on render because it affects the performance.